### ISSUE TITLE
Искусственный студент

### ISSUE THREAD

**Issue by sherdim** [neuro-sfedu/mcp-fs#23]
Идея из DeepTutor #19 - для тестирования и прогона от лица LLM. 

У них в TutorBench профиль - это образование, цель обучения и уровень владения по темам, плюс пробелы трёх типов - всё из знаниевой плоскости. 

У нас - карта знаний + состояние (влияние на мотивацию) + генератор ЭЭГ и пульса + SOUL.md (психологический портрет)

ФС с вариацией от времени суток добавит новую ось в диаграмму состояний. 

Вопросы:
- как связывать ФС с картой знаний: 
  - или простая регрессия: усталость просто снижает демонстрируемый уровень знаний, 
  - или выборочно выключает сложные топики
  - или отключает старые знания, сохраняя свежие

- как ФС влияет на поведение - темп, длина ответов, готовность продолжать.

- включать ли социальные предпочтения в выборе персоны тьютора? Тема интересная, но скользкая (отношения полов, манипуляции)

**Comment by sherdim** [comment]
Для демонстрации должно хватить 4 профилей (по опыту определения психотипов):
- типичный студент, слегка ленивый, но вменяемый, хорошист
- активный талантливый, слегка восторженный (непуганый), отличник
- заострённый проницательный, но непредсказуемый, невротичный, перекошенный по знаниям
- тугой, осторожный, пассивный, троечник, требующий стимуляции


Интеллект при этом одинаковый.

t: 20260906

**Comment by dmi-rriy** [comment]
Посмотрел, как искусственный студент сделан в LEA - там уже есть модель с утомлением, можно не с нуля.

Что там есть

Профиль в simulation/lea_simulation_framework.py, класс LearnerProfile: тип личности плюс 13 параметров - learning_rate, initial_motivation, cognitive_capacity (порог когнитивной нагрузки), attention_span, help_seeking, persistence, metacognition, accuracy_variance, time_per_interaction, fatigue_rate, question_types_preferred, scaffolding_preference, feedback_sensitivity.

Личностей шесть: confident_quick, methodical_steady, struggling_persistent, easily_distracted, perfectionist, creative_divergent.

По вопросу про связь ФС с картой знаний

У них первый вариант, но не линейный, а мультипликативный. Вероятность правильного ответа:

success_prob = base_success * difficulty_factor * cl_factor * motivation_factor * fatigue_factor + zpd_bonus + momentum

где fatigue_factor = 1 - fatigue_level. То есть усталость не выключает темы выборочно, а равномерно давит вероятность.

Тут важная деталь: base_success у них это learning_rate из профиля, а не текущее владение темой. То есть карта знаний в эту формулу вообще не входит - способность учиться есть, а что именно студент уже знает, на вероятность ответа не влияет. По-моему это их слабое место, у нас так делать не стоит.

Само утомление копится от нагрузки:
fatigue_increase = (0.02 + cognitive_load/100) * profile.fatigue_rate

Скорость утомления - черта личности, прирост зависит от текущей нагрузки.

По вопросу про влияние на поведение

Утомление влияет не только на правильность, но и на продолжение: при fatigue_level > 0.8 студент уходит. Отдельно порог по мотивации (< 0.2 тоже уходит) и лимит по attention_span. Есть механика перерыва - после него утомление падает на случайную величину от 0.1 до 0.3.

Ещё есть momentum: несколько верных подряд дают бонус до +0.2, несколько неверных штраф до -0.3. Это ближе к мотивации, чем к знаниям, и нам полезно - серия неудач гасит сильнее, чем одна.

Что взял бы и что поменял

Взял бы структуру профиля и формулу с fatigue_factor, она простая и рабочая. Но base_success надо считать от владения темой, а не от learning_rate.

Главное отличие: у них fatigue это чисто вычисляемое число, растёт только от нагрузки внутри сессии. У нас генератор ЭЭГ и пульса с вариацией от времени суток - утомление приходит извне и может быть высоким уже на старте. У LEA студент устаёт только от занятия, у нас может прийти уставшим. По-моему это и есть наша новизна.

Про третий вариант из карточки (отключает старые знания, сохраняя свежие) - может, наоборот? Свежевыученное хуже консолидировано и должно страдать первым, а старое усвоенное держаться дольше. Но тут я не уверен, надо смотреть литературу.

Про четыре профиля - согласен, хватит. Ваши четыре ложатся на их параметры почти один в один: хорошист это methodical_steady, отличник confident_quick, невротичный easily_distracted с высоким accuracy_variance, троечник struggling_persistent с низкой мотивацией.

Про социальные предпочтения в выборе персоны - на этом этапе не стал бы. На проверку методики не влияет, а рисков добавляет.

Вдобавок: разбирал вместе с ИИ (Claude), просил найти в LEA как устроен симулятор студента и как утомление влияет на ответы, потом перепроверял формулы по коду сам. Так что это вайб-разбор, но цифры и строки проверены.

**Comment by sherdim** [comment]
мда, это ж как во всех этих persistence, metacognition разбираться...

модель хорошая! [Sophisticated](https://github.com/tep00018/LEA/blob/5e99380df83e973b5bb3867a7f300f1e8d1b588a/simulation/lea_simulation_framework.py#L54) !!!
и анализы comprehensive!

надо бы и саму диссертацию читать, как анализы преподносятся (интерпретируются)...

например, эта модель 7-факторная авторская?

    learning_rate: float          # 0.1-0.9 (how quickly they learn)
    initial_motivation: float     # 0.3-0.9 (starting motivation)
    cognitive_capacity: float     # 3.0-8.0 (cognitive load tolerance)
    attention_span: float         # 0.3-0.9 (ability to maintain focus)
    help_seeking: float          # 0.2-0.8 (likelihood to ask for help)
    persistence: float           # 0.3-0.9 (how long they keep trying)
    metacognition: float         # 0.2-0.8 (self-awareness of learning)

уход в сильные синтетические симуляции чреват тем, что потом сложно сравнить с реальными студентами, у которых эти факторы неясно как измерить (вернее ясно, но трудоемко и ненадежно)

у Teri (верим, что она сама долго делала?) расчеты подозрительно [простые](https://github.com/tep00018/LEA/blob/5e99380df83e973b5bb3867a7f300f1e8d1b588a/src/core/agent_orchestrator_mcp.py#L423) - пороговые значения не настраиваемые. Пример:

        if (metrics.persistence_level == PersistenceLevel.EARLY_ABANDONMENT or
            metrics.persistence_level == PersistenceLevel.DECREASED_PERSISTENCE or
            metrics.affective_score < -0.3 or
            metrics.performance_score < 0.5):
            return MotivationState.MOTIVATION_DROP

про усталость:
в симуляции и у них захардкоден исходный [0](https://github.com/tep00018/LEA/blob/5e99380df83e973b5bb3867a7f300f1e8d1b588a/simulation/lea_simulation_framework.py#L118) ,
но может такие условия симуляции.

Проверка усталости на входе - проблема типа "допускового контроля", но если идет постоянный мониторинг, то можно начинать симуляцию с любого состояния и делать прогноз: например, виртуально сравнить учебу "ночью в один присест" и учебу урывками.
И контроль состояния действительно снимает проблему допуска: под любое состояние можно подобрать полезную нагрузку.


по типам (которые не паруются с предложенными мной выше):
- struggling_persistent - тугой, но старательный - неправдоподобно. Кто-то знает таких?
- "easily_distracted"  # Variable performance, needs engagement - ADHD?  можно рассмотреть, но скорее это пара к тугому троечнику - он тоже отвлекается (но без патологии). При ADHD буквально не может сосредоточиться.
- "perfectionist"  - интересный тип, вроде гений, но статистически таких нет . учесть можно, но проверить не сможем.

**Comment by sherdim** [comment]
Про третий вариант из карточки - отдельный комментарий! (типа педагогический)

показательный важный момент - в какой-то момент обсуждения появляются противоположные гипотезы.
это то самое - где возникает стимул провести экспериментальную проверку.
на поверхности обычно все согласны, а вот глубинные нюансы - предмет научных споров, хотя для общей картины может показаться - предмет спора несущественный.

Какие варианты в данном примере:
1. Эту проблему кто-то уже формулировал, и есть работа и статья. Можно повторить для проверки, может поменять условия, но это будет проверка уже известной гипотезы.
2. Гипотеза новая: "ФС действует через механизмы памяти, частично их отключая, что приводит к разной доступности усвоенных навыков"
    2.1  ограничение через ресурсы - они уменьшаются при усталости, грубо - что-то одно только в голове (последнее, свежее) - сравнения с альтернативами отключены. Например, боксер с опухшим мозгом на 3-м раунде, просто пытается сделать то, что кричит тренер, а весь прежний арсенал как-будто забыл.
   2.2  Свежевыученное хуже консолидировано и должно страдать первым, а старое усвоенное держаться дольше. Правдоподобно, но на большем мастшабе - например, вчерашний урок уже твердо усвоил и используешь, а сегодняшний - еще путаешься и от усталости - сплошной "винегрет" в голове.

- может оказаться, что обе гипотезы применимы, но для разных временных окон: суточные временные окна - одна зависимость, минутные временные окна - другая. то есть объекты рассмотрения разные, надо просто уточнять временную шкалу явлений.
- может оказаться, что обе гипотезы не работают: забывание происходит само собой, а усталость одинаково действует на снижение эффективности и старых и новых навыков/знаний.


Как проверить? надо заложить в методику все необходимые контроли.

Единица - исследовательская задача (ИЗ)

тестовые ИЗ надо провести в условиях хорошего ФС и усталости.
в каждой ИЗ должны требоваться навыки, которые изучены ранее:  Пронумеруем навыки убывающими индексами (-1... -10) недавние, а -101...-110 - вчерашние или еще раньше. Например $A1$ -5 мин,   $A2$  -10 мин, $A6$  -60 мин от текущего времени.
Тогда надо 
- чтобы оценить эффект от быстрого вытеснения из памяти сравнить результаты $P(A \mid \{A1,A3, A101\})$ и $P(A|{A7,A9, A101})$ .
- чтобы оценить эффект от консолидации памяти сравнить результаты $P(A \mid \{A1,A3, A7\})$ и $P(A|{A1,A104, A105})$ .

Поскольку задачи разные, то их надо примерно выровнять по сложности, задать пределы забывания, 
в идеале сделать простые повторные тесты для чистого эффекта забывания одного навыка (т.е. дублировать задачу но с другими данными)  
$P(A \mid A4)$ , $P(A \mid A3)$  
vs. $P(A \mid A8)$ и $P(A \mid A10)$
vs. $P(A \mid A104)$ и $P(A \mid A105)$

Потом еще наделать таких задач много, сбалансировать, псевдослучайно замешать и в реальных экспериментах предъявлять в двух разных порядках для контроля влияния предъявления.

**Comment by dmi-rriy** [comment]
Отвечу на оба комментария сразу.

ПРО ФАКТОРЫ

Нашёл саму статью - это не только магистерская, есть публикация на ICAART 2026, стр. 512-523, четыре автора: Rumble, Zarrin, Lovell, Falconer. PDF открытый:
https://www.scitepress.org/Papers/2026/144189/144189.pdf

Модель не авторская. В статье написано, что профили adapted from personality-aware learner modeling frameworks со ссылкой на Yuan et al. 2025 (arXiv 2508.05622).

Но код и статья расходятся. В статье (Table 1) шесть профилей - Advanced Eager, Reflective, Struggling Eager, Capable Disengaged, Novice Variable, Inconsistent Adaptive - и пять параметров: Prior Knowledge, CL Tolerance, Motivation, Engagement Style, Learning Rate. В коде шесть других типов и 14 полей в LearnerProfile, те семь что вы процитировали - первые семь из четырнадцати. То есть в публикации этой модели нет, код развивался после статьи. Ваше замечание по struggling_persistent и perfectionist относится именно к коду.

Теория у них на уровне архитектуры, не симулятора: Cognitive Load Theory (Sweller, Paas), ЗБР и конструктивизм (Wood 1976, Выготский 1978), Self-Determination Theory (Ryan и Deci 2000), Bandura 1997 и Zimmerman 2002.

По порогам вы были правы, авторы признают напрямую. Формула нагрузки:
CL = β0 + β1*G + β2*(1−A) + β3*Q + β4*Ttype
β0=0.5, β1=1.0, β2=4.5, β3=0.1, β4=1.0. Дословно: коэффициенты определены офлайновыми симуляциями Монте-Карло и могут не отражать оптимальных значений.

Там же про усталость в основной системе: Q это число непрерывных взаимодействий с коэффициентом 0.1. То есть усталость сведена к счётчику реплик и это самый маленький коэффициент в формуле.

Главное: в Limitations авторы пишут, что подсистема памяти логирует состояния, но не извлекает исторические паттерны для будущих взаимодействий. Это то же, что мы нашли в коде в #12. Первым пунктом Future Work у них Active Memory Utilization. Разомкнутый цикл - признанное ограничение.

По опасению про синтетику: у них есть продолжение, arXiv 2607.13370, июль 2026, Cross-Course Scalability and Classroom Evaluation - там реальный класс. Ещё не читал.

ПРО ГИПОТЕЗЫ

По варианту 1 - обе ваши гипотезы уже сформулированы, но в разных областях и под разными именами.

2.1 (ресурсы уменьшаются) - это resource depletion framework. Утомление ограничивает доступный объём рабочей памяти, отсюда эффект боксёра.

2.2 (свежее хуже консолидировано) - systems consolidation. Свежий след уязвим к интерференции, пока не закрепился.

То есть это не спор двух гипотез, а два механизма из разных областей. Ваша догадка про временные окна в целом подтверждается, но граница не такая резкая: в работах по консолидации эффект от того, чем человек занят сразу после заучивания, виден уже через 12-24 минуты и сохраняется через неделю. То есть минутная шкала это не только рабочая память, консолидация туда уже попадает.

Нашёл работу прямо по вашему боксёрскому примеру - Scientific Reports 2026, бадминтон, 36 новичков:
https://www.nature.com/articles/s41598-026-40994-6
Два полезных факта. Рабочую память мерили через n-back, при ментальном утомлении упала с 38.0 до 20.2. И ментальное утомление било сильнее физического и смешанного - значит для генератора ЭЭГ и пульса тип утомления надо различать, а не сводить к одной шкале.

Ещё нашёл вариант, которого нет в вашем списке. Индивидуальные различия в объёме рабочей памяти модерируют эффект консолидации, причём у людей с низким объёмом он разворачивается в обратную сторону:
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7578020/
То есть возможен пятый исход: обе гипотезы верны, но для разных людей. Тогда усреднение по группе может дать ноль просто потому, что эффекты в подгруппах гасят друг друга.

ПРО МЕТОДИКУ

Логика сравнений понятна: в первом контроль через A101 в обоих наборах, во втором через A1. Три замечания.

Первое. Раз объём рабочей памяти модерирует эффект, испытуемых стоит стратифицировать по нему до начала, иначе рискуем получить ноль на усреднении. Нужен базовый замер объёма рабочей памяти - в той работе использовали не n-back, а задачи на запоминание с параллельной нагрузкой, это разные инструменты и выбирать надо осознанно.

Второе. В вашей шкале разрыв между -60 мин и вчерашним. Судя по литературе, консолидация разворачивается уже в первые десятки минут и продолжается сутками, так что интересный переход как раз внутри этого разрыва. Точка на 3-6 часов помогла бы увидеть, где механизм меняется, а не только два края.

Третье, и это скорее вопрос. Гипотеза про механизмы памяти на синтетических студентах в принципе не проверяется - мы получим ровно то, что сами заложили в модель. Симулятор годится отладить протокол, сбалансировать задачи и оценить нужное число проб, но саму гипотезу придётся проверять на живых. Это не возражение к методике, а вопрос, на каком этапе мы переходим к людям.

Оговорюсь: по памяти читал аннотации и выжимки, полные тексты не смотрел. Если направление рабочее, могу разобрать протокол из бадминтонной статьи подробнее - там замеры до и после, близко к тому, что вы описали.

Вайб-разбор: статьи искал через Claude, PDF по LEA читал и сверял с кодом сам.

**Comment by dmi-rriy** [comment]
По вопросу про усталость и знания. Смотрел формулу в lea_simulation_framework.py:314-345:

success_prob = base_success * difficulty_factor * cl_factor * motivation_factor * fatigue_factor + zpd_bonus + momentum
fatigue_factor = 1.0 - fatigue_level

Это простая регрессия — плоский множитель, одинаковый для любого вопроса. Ни выборочного отключения сложных топиков, ни сохранения свежих знаний при забывании старых там нет — и не потому что не заметили, а потому что тематической разбивки знаний в этом модуле вообще нет: _update_topic_mastery() хардкодит current_topic = "general" (строка 478, комментарий автора «Would be more specific in real implementation»), всё падает в один бакет.

При этом в другой части LEA (go_strategy.py) mastery действительно считается по темам (per-GO). Но этот модуль и симуляция друг друга не видят — GO-mastery не знает про fatigue, формула успеха не знает про GO. Грануляция по темам и усталость у LEA есть, но раздельно.

По темпу/готовности продолжать — механизм есть, разбирали в #19: _choose_interaction_type() смещает вероятности к take_break/request_help, когда cognitive_load > 7 или motivation < 0.4.

Отдельно, не по коду, а по существу третьего пункта (про персону/социальные предпочтения): я бы не стал строить персону на демографии. Гендер плохо предсказывает индивидуальный отклик на стиль обучения — разброс внутри группы обычно больше разницы между группами, так что это даст менее точную персонализацию, чем то, что уже есть (motivation_state, cognitive_load, ZPD). И это именно та грань, которую вы сами назвали манипуляцией — подстройка тона под предполагаемые гендерные ожидания, а не под конкретного человека. Для проекта под названием «Антропоцентрический ИИ» это репутационно рискованно, если всплывёт в статье. У нас и данных по своей выборке нет, чтобы это обосновать. Расширять персонализацию — да, но по измеримым индивидуальным сигналам, а не по демографии.

**Comment by ilya-lys** [comment]
@dmi-rriy подключу сюда наработку с 19й задачи - там разбирали, куда DeepTutor вообще встраивает поведенческий сигнал (Ds - история, Dw - диагностика слабых мест, Dr - саморефлексия тьютора), и по абляции Fitness (калибровка сложности) держится именно на связке Dw -> question-generator, а не на Writer/Dr.

Смотрю на твою находку по fatigue_level в LEA и на то, что _update_topic_mastery хардкодит topic="general" - то есть текущий сигнал усталости плоский, без привязки к теме. А тематическая гранулярность в go_strategy.py наоборот есть, просто эти два модуля друг друга не видят.

Смысл: если наша ФС в итоге должна питать question-generator (как в DeepTutor), ей нужна тематическая привязка - иначе нечем будет калибровать сложность по конкретной теме, а не "в среднем по студенту". То есть даже если механизм (resource depletion / consolidation) экспериментально подтвердится, сигнал физически нельзя будет подключить туда, где он нужен по архитектуре, пока усталость не связана с темой так же, как mastery в go_strategy.
